### PR TITLE
fix(input-time-picker): automatically close when clicked outside

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #8639

## Summary

- add autoClose to internal popover
- add test
- update referenceElement to use element rather than id

### CODE REVIEW

**Comment by eriklharper** [comment]
I'm noticing some undesirable behavior when toggling the caret in `input-time-picker`.  The popup doesn't seem to open at all when clicking the caret, but it seems to work well when just focusing the text input and then blurring it.

**Comment by driskull** [comment]
@eriklharper checked in a fix. Can you review again?

**Review by eriklharper — APPROVED** [review]
> @eriklharper checked in a fix. Can you review again?

Heck yeah, seems to be working fine now :)

File: packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx [hunk 1]
```diff
@@ -1073,7 +1068,7 @@ export class InputTimePicker
 
   renderToggleIcon(open: boolean): VNode {
     return (
-      <span class={CSS.toggleIcon}>
+      <span class={CSS.toggleIcon} slot="action">
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx:1071]: Nice!

**Review by jcfranco — APPROVED** [review]
🚀

**Pull request merged**